x-pack/libbeat/management/manager.go

```diff
@@ -217,10 +218,7 @@ func (cm *Manager) SetPayload(payload map[string]interface{}) {
 }
 
 func (cm *Manager) OnStop() {
-	if cm.stopFunc != nil {
-		cm.client.Status(proto.StateObserved_STOPPING, "Stopping", nil)
-		cm.stopFunc()
-	}
+	cm.client.Status(proto.StateObserved_STOPPING, "Stopping", nil)
```

Comment:
How does the Elastic Agent no signal a beat to stop? That comes from the control protocol. The Elastic Agent expects that once a beat receives a STOP that it should stop itself, only after a period of time (if it doesn't) does it force kill it.

I think this change here will really break that behavior and is incorrect.

Why the removal of the `stopFunc`?

Reply:
I will take a look, when I've moved the method and the flow of operation I thought the stopFunc was superfluous, but I will double check it.

Reply:
@blakerouse I've looked a the code, and from what I see I believe Elastic Agent was always doing a force quit of the process, the `stopfunc()` was never configured. AFAIK, allowing a graceful shutdown on the beats is a big task, because they don't have any common shutdown flow. I am not sure this is worth the effort before refactoring the inputs into smaller components.

Reply:
Note: I think this is a bug, but require a bigger effort and it's more risky.